demos/heat_equation/setup.py

```diff
@@ -0,0 +1,53 @@
+from firedrake import *
+from firedrake_adjoint import * 
+
+
+def initial_mesh():
+    """
+    Construction of initial mesh
+    """
+    return RectangleMesh(5, 5, (-1,1), (-1,1))
```

Comment:
`initial_mesh` needs to accept an input parameter `n` that determines the mesh resolution. How about this?
```
def initial_mesh(n):
    """
    Construction of initial mesh
    """
    return RectangleMesh(5 * 2 ** n, 5 * 2 ** n, (-1,1), (-1,1))
```